tests/backup/backup_kdmp_test.go

```diff
@@ -1400,3 +1401,158 @@ var _ = Describe("{ExcludeInvalidDirectoryFileBackup}", func() {
+			rand.Seed(time.Now().UnixNano())
+			// Randomly delete backups from the backup list
+			for i := 0; i < backupDeleteCount; i++ {
+				backupIndexToDelete := rand.Intn(len(backupNameList))
+				backupUID, err := Inst().Backup.GetBackupUID(ctx, backupNameList[backupIndexToDelete], orgID)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Getting backuip UID for backup %s", backupNameList[backupIndexToDelete]))
+				_, err = DeleteBackup(backupNameList[backupIndexToDelete], backupUID, orgID, ctx)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Verifying backup deletion - %s", backupNameList[backupIndexToDelete]))
+				backupNameList = append(backupNameList[:backupIndexToDelete], backupNameList[backupIndexToDelete+1:]...)
+			}
+			log.Infof("List of remaining backups - %v", backupNameList)
+		})
+
+		Step("Restoring random backups on destination cluster", func() {
+			log.InfoD("Restoring %v random backups on destination cluster", backupRestoreCount)
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Unable to fetch px-central-admin ctx")
+			// Shuffle the existing list
+			for i := len(backupNameList) - 1; i > 0; i-- {
+				j := rand.Intn(i + 1)
+				backupNameList[i], backupNameList[j] = backupNameList[j], backupNameList[i]
+			}
+			backupListForRestore = backupNameList[:backupRestoreCount]
+			log.Infof("List of backups to be restored - %v", backupListForRestore)
+			for _, backupName := range backupListForRestore {
+				appContextsExpectedInBackup := FilterAppContextsByNamespace(scheduledAppContexts, bkpNamespaces)
+				restoreName := fmt.Sprintf("%s-%v", restoreNamePrefix, RandomString(10))
+				err = CreateRestoreWithValidation(ctx, restoreName, backupName, make(map[string]string), make(map[string]string), destinationClusterName, orgID, appContextsExpectedInBackup)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore [%s] from backup [%s]", restoreName, backupName))
+				restoreNames = append(restoreNames, restoreName)
+			}
+		})
+	})
+
+	JustAfterEach(func() {
+		defer EndPxBackupTorpedoTest(scheduledAppContexts)
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		log.InfoD("Deleting the restores")
+		for _, restoreName := range restoreNames {
+			err = DeleteRestore(restoreName, orgID, ctx)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Deleting restore [%s]", restoreName))
+		}
+		log.InfoD("Deleting the deployed apps after the testcase")
+		opts := make(map[string]bool)
+		opts[SkipClusterScopedObjects] = true
+		DestroyApps(scheduledAppContexts, opts)
+		log.InfoD("Deleting the px-backup objects")
+		CleanupCloudSettingsAndClusters(backupLocationMap, cloudAccountName, cloudCredUID, ctx)
+		log.InfoD("Switching context to destination cluster for clean up")
+		err = SetDestinationKubeConfig()
+		log.FailOnError(err, "Unable to switch context to destination cluster [%s]", destinationClusterName)
+		DestroyApps(scheduledAppContexts, opts)
```

Comment:
We dont have to do this. In `EndPxBackupTorpedoTest` we have a function called `DeleteAllNamespacesCreatedByTestCase` which will cleanup all the apps.